yeonjae/baekjoon/구현/2108-통계학.js

```diff
@@ -0,0 +1,43 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const answer = [];
+  const numList = input
+    .slice(1)
+    .map(Number)
+    .sort((a, b) => a - b);
+
+  //   산술평균
+  const average = numList.reduce((acc, cur) => acc + cur, 0);
+  answer.push(Math.round(average / numList.length));
+
+  // 중앙값
+  const centerLocation = numList[Math.floor(numList.length / 2)];
+  answer.push(centerLocation);
+
+  //최빈값
+  const obj = {};
+  numList.forEach((el) => (obj[el] = (obj[el] || 0) + 1));
+
+  const max = Math.max(...Object.values(obj));
+  const mode = Object.entries(obj)
+    .filter((el) => el[1] === max)
+    .sort((a, b) => a[0] - b[0]);
+
+  answer.push(mode.length !== 1 ? +mode[1][0] : +mode[0][0]);
+
+  //최댓값 최솟값차이
+  const range = Math.abs(Math.max(...numList) - Math.min(...numList));
+  answer.push(range);
+
+  console.log(answer.join("\n"));
```

Comment:
```suggestion
  const answer = [];
  const numList = input
    .slice(1)
    .map(Number)
    .sort((a, b) => a - b);

  //   산술평균
  const average = numList.reduce((acc, cur) => acc + cur, 0);
  answer.push(Math.round(average / numList.length));

  // 중앙값
  const centerLocation = numList[Math.floor(numList.length / 2)];
  answer.push(centerLocation);

  //최빈값
  const obj = {};
  numList.forEach((el) => (obj[el] = (obj[el] || 0) + 1));

  const max = Math.max(...Object.values(obj));
  const mode = Object.entries(obj)
    .filter((el) => el[1] === max)
    .sort((a, b) => a[0] - b[0]);

  answer.push(mode.length !== 1 ? +mode[1][0] : +mode[0][0]);

  //최댓값 최솟값차이
  const range = Math.abs(Math.max(...numList) - Math.min(...numList));
  answer.push(range);

  console.log(answer.join("\n"));
```
- 각 로직들을 주석으로 분리하는 것 보다 각각의 함수로 빼서 활용하는게 어떨까요? 윤우님 코드처럼 객체의 메서드로 작성해서 활용하는 것을 추천드립니다!
- 그리고 numList.length와 같은 값들은 입력값으로 n으로 주어지기 때문에 이를 활용해주세요!
- 그외 최빈값 관련해서 피드백은 윤우님과 동일하니 윤우님 코멘트 확인해주세요!

Reply:
저번처럼 활용하는걸 생각을 못했네요!ㅠㅠ 넵 감사합니다